Lora: Document all possible error codes for LoRaWANBase methods

kivaisan opened the pull request:
### Description

Document all possible error codes for LoRaWANBase methods.

Also duplicate documentation is removed from LoRaWANInterface which
inherits LoRaWANBase class.


### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Feature
    [ ] Breaking change

Commit:
Lora: Document all possible error codes for LoRaWANBase methods

Also duplicate documentation is removed from LoRaWANInterface which
inherits LoRaWANBase class.

kivaisan commented:
@hasnainvirk, @AnttiKauppila, @kjbracey-arm, please review.

AnttiKauppila reviewed (approved):
LGTM

Commit:
Copy edit LoraWANBase.h

Copy edit file, mostly for consistent U.S. spelling and precise language.

AnotherButler commented on features/lorawan/LoRaWANBase.h:
```diff
@@ -52,15 +82,44 @@ class LoRaWANBase {
-     * @return    For ABP:  If everything goes well, LORAWAN_STATUS_OK is returned for first call followed by
+    /** Connect OTAA or ABP with parameters
+     *
+     * All connection parameters are chosen by the user and provided in the
```
Note: Sometimes we say "you," and sometimes we say "the user". If the reader is the user, it should always be "you". If the reader's not the user, who is the reader?

Commit:
Lora: Reformate LoRaWANBase documentation

- Modified "user" to "you"
- Modified line length to ~100 characters.

kivaisan commented:
Please review again

Commit:
Copy edit LoRaWANBase.h

Copy edit file for minor grammar nits.

cmonr reviewed (commented):
:+1:

cmonr commented:
/morph build

Commit:
Lora: send and receive methods return LORAWAN_STATUS_NOT_INITIALIZED if not initialized

In order have a consistent return value for all methods in case of system is uninitialized
now also send and receive methods can return LORAWAN_STATUS_NOT_INITIALIZED.

kivaisan commented:
@hasnainvirk @AnttiKauppila please review again.

hasnainvirk commented:
Looks good.

cmonr commented:
Good small change. @AnttiKauppila, a small PR update, if you would like to re-review.

/morph build

cmonr commented:
Strange export build error.

/morph export-build

AnttiKauppila commented:
This can be merged, looks good.
